### ISSUE TITLE
[Feature Request] laminarc dequeue <<job Name>>

### ISSUE THREAD

**Issue by stephengaito** [ohwgiles/laminar#128]
Hello, 
I am currently running a complex system of cascaded jobs, in a context with a limit of two jobs running simultaneously (anything more would bring my desktop to a halt).

Often, I get into a situation where I find I would like to abort the whole cascade of jobs, but since a number of them are in the queued state, I do not have the ability to remove them until I have aborted enough of the currently running tasks for the queue to empty (since only *running* jobs can be aborted).

**Would it be possible to add a "laminarc dequeue" command?**

Regards, 
Stephen Gaito

**Comment by ohwgiles** [comment]
I've been thinking about another issue that may result in a fix for this one.

Currently, a run number is only assigned once a job comes **off** the queue. This means that an upstream job has to use `laminarc start` rather than `laminarc queue` if it wants to maintain a record in the log of which downstream run was started. Of course `laminarc start` blocks until the job can actually start, which might be a problem if the parent job is taking up some necessary resource. To fix this, I'm thinking of assigning run numbers when a job is put **onto** the queue instead.

This would then make it easy for `laminarc abort` to work on queued jobs too. Sound good?

**Comment by stephengaito** [comment]
Yes I think that sounds very good.

My particular use case (at the moment) is to use Laminar to coordinate the build/test and ultimate release of a complex project across 4 different OS/Releases. As such I have a number of parameterized jobs which will often be queued all at the same time. Having a run number at queue 
time, would help distinguish these multiple jobs.

I am assuming that once a run number has been assigned, it would be much easier to abort that job (even if still queued).

Again, this is a very nice tool! Many thanks for your work!

**Comment by stephengaito** [comment]
For the moment, I have created the following (crude) shell script which slowly walks through the list of running jobs aborting each one it finds... so (for the moment)... I do have a work-around....
```
#!/bin/bash

# This Laminar bash script aborts all running Laminar jobs

tryAgain=true

while $tryAgain ; do
  tryAgain=false
  for aJob in $(laminarc show-running) ; do
    jobName=$(echo $aJob | cut -d':' -f1)
    jobNum=$(echo $aJob | cut -d':' -f2)
    echo laminarc abort $jobName $jobNum
    laminarc abort $jobName $jobNum
    tryAgain=true
    echo "."
    sleep 1
  done
  echo ".."
  sleep 5
done
```

**Comment by ohwgiles** [comment]
This should be implemented with 3c7433e71e9cad2fa1589b81a603663e1bbb06d9. It would be great if you could try out the latest master and see if it works well for you.
